projects/pureth-eip-7919-erigon.md

```diff
@@ -0,0 +1,173 @@
+    block_number: uint64
+    block_hash: Root
+    timestamp: uint64
+    dummy_value: uint64  # 2**64-1
+```
+
+### Consensus Changes
+
+replace the `logs_bloom` field of the header schema with `log_index_root` which is the root hash of the LogIndex structure after adding the logs emitted in the given block.
+
+## Filtermaps Components and Flow
+
+### Overview
+
+- **Matcher:** Matches requested log values using multi-layer hash mapping.
+- **Indexer:** Indexes events sequentially for rapid lookups.
+- **Filtermaps:** Multi-layered structure enables efficient queries and proof generation.
+
+### Flowchart
+
+```mermaid
+flowchart TD
+    A[Start_Query]
+    B[Validate_Block_Range]
+    C[Map_Log_Index_Range]
+    D[Hash_Log_Value]
+    E[Traverse_Filtermap_Layers]
+    F[Record_Matches]
+    G[Pattern_Logic]
+    H[Verify_Logs]
+    I[Return_Results]
+
+    A --> B
+    B --> C
+    C --> D
+    D --> E
+    E --> F
+    F --> G
+    G --> H
+    H --> I
+```
+
+### Multi-layer Mapping Logic
+
+```go
+for layer := 0; layer < maxLayers; layer++ {
+    idx := hashFNV(value, layer) // position in layer’s row
+    entry := row[layer][idx]     // 24-bit number or marker
+    if entry == 0 {
+        break // no match
+    }
+    if entry < threshold {
+        recordPotentialMatches(layer, idx, entry)
+        break
+    }
+    // entry >= threshold, go deep
+}
+```
+
+### Pattern Matching and Verification
+
+- **matchAnyProver:** Logical OR across topic values.
+- **matchSequenceProver:** Logical AND for specific topics.
+- **False Positive Filtering:** Final step checks actual log contents via Merkle proofs.
+
+### Proof System
+
+- Compact binary proofs encode query, roots, layer data, and Merkle log proofs.
+- Partial rows handled by descending to finer layers.
+- Test vectors in `.proof` and `.json` formats validate correctness.
+
+## Roadmap
+
+**Milestone 0**
+- **Weeks 6–7:** Familiarize with Erigon and start designing filtermaps specification.
+
+**Milestone 1**
+- **Weeks 8–10:** Implement mathematical utilities, map renderer, and filtermaps data structure. Write initial unit tests.
+
+**Milestone 2**
+- **Weeks 11–16:** Build indexer, matcher, matcher backend. Expand test coverage.
+
+**Milestone 3**
+- **Weeks 17–20:** Implement proof generation and verification mechanisms.
+
+**Milestone 4**
+- **Weeks 20–21:** Test thoroughly using Hive, Kurtosis, and RPC endpoints. Develop custom and integration tests.
+
+**Milestone 5**
+- **Week 21+:** Implement related EIPs (e.g., EIP-7708, EIP-7799) or pursue further performance optimizations.
+
+## Challenges
+
+- Adapting complex EIP logic
+- Some library issues or differences between geth and erigon.
+
+## Goals
+
+- **Complete working filtermaps integration** in Erigon.
+- **Verifiable, efficient log queries**.
+- Comprehensive testing for the code implemented.
+
+## Mentor 
+
+[Etan Kissling](https://github.com/etan-status)
+
+## Resources
+https://pureth.guide
+https://web3-purrified.box
```

Comment:
https://purified-web3.box would be the correct link here, if you want to refer to the prior work with EthereumJS